### PR TITLE
 Feature Noise XKpsk3 Move topology control into its own crate

### PR DESCRIPTION
**Pull request by simonwicky**
# Description
This is a part of stacked PRs to integrate XKpsk3 Noise connection between nym nodes

This PR concerns moves topology control into its own crate, removing the client-core import in mix nodes and gateways

https://github.com/nymtech/nym/pull/4360 -> https://github.com/nymtech/nym/pull/4361 -> https://github.com/nymtech/nym/pull/4373 (this one) -> https://github.com/nymtech/nym/pull/4362 

# Checklist:

- [ ] added a changelog entry to `CHANGELOG.md`

### CODE REVIEW

File: common/topology-control/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[package]
+name = "nym-topology-control"
+version = "1.1.15"
```
Comment (octol) [inline common/topology-control/Cargo.toml]: Probably leave version at 0.1.0

File: common/topology-control/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+authors = ["Dave Hrycyszyn <futurechimp@users.noreply.github.com>"]
```
Comment (octol) [inline common/topology-control/Cargo.toml]: Basically most of these fields should probably use the workspace defaults

**Review by octol — APPROVED** [review]

File: common/topology-control/Cargo.toml (hunk 1 above)
Comment (simonwicky) [reply common/topology-control/Cargo.toml]: Applied in 384cafc

File: common/topology-control/Cargo.toml (hunk 2 above)
Comment (simonwicky) [reply common/topology-control/Cargo.toml]: Apply in 384cafc

**Pull request merged**